### PR TITLE
Feature/add character

### PR DESCRIPTION
**Pull request by RomanDevelop**
- [x] 1. Create Add Character Page, with similar structure that we have in the project / 21:57 talking about new feature /
- [x] 2. Add Method in cubit (addCharacter)
- [x] 3. Add Method in service and call it from your cubit

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -53,4 +53,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 70d9d25280d0dd177a5f637cdb0f0b0b12c6a189
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.14.3
```
Comment (arrrtem22) [inline ios/Podfile.lock:56]: On real project we usually do not touch config files, just keep in mind that

File: lib/common/network/api/characters_api.dart [hunk 2]
```diff
@@ -10,18 +11,42 @@ part 'characters_api.g.dart';
...
+      FakeApi.characters.isEnabled ? FakeCharactersApi() : _CharactersApi(dio);
 
   @GET('')
   Future<Characters> getCharacters();
+
+  @POST('')
+  Future<Characters> addCharacter();
 }
 
 class FakeCharactersApi implements CharactersApi {
```
Comment (arrrtem22) [inline lib/common/network/api/characters_api.dart:24]: let's create variable in FakeCharactersApi for ex fakeCharacters and work with it, in getCharacters we'll return it, in addCharacter we'll add something

File: lib/common/network/fake_api_manager.dart [hunk 3]
```diff
@@ -14,8 +13,7 @@ enum FakeApi {
 
 class FakeApiManager {
   static final _enabledFakeApi = <FakeApi>{
-    //Place here fake api that should be enabled by default
-    // if (kDebugMode) FakeApi.characters,
+    if (kDebugMode) FakeApi.characters,
```
Comment (arrrtem22) [inline lib/common/network/fake_api_manager.dart]: let's revert this changes, app in develop and master branch should work with real api (but add character feature won't work due to api luck)

File: lib/common/network/api/characters_api.dart [hunk 4]
```diff
@@ -10,18 +11,42 @@ part 'characters_api.g.dart';
...
+
+    final fakeResponse = Characters(characters: fakeCharacters);
+
+    await Future.delayed(const Duration(seconds: 2));
+
+    return fakeResponse;
+  }
+
+  @override
+  Future<Characters> addCharacter() async {
```
Comment (arrrtem22) [inline lib/common/network/api/characters_api.dart]: why do we return Future<Character> here, but here https://github.com/arrrtem22/character_viewer/pull/11/files#diff-7b7f72fa6f5b258c8531378615b98142f95fd2a88ea3bfd9b67d2d33064d7c46R14 we return nothing

File: lib/feature/add_character/add_character_page.dart [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      listener: (context, state) {},
+      builder: (context, state) {
+        return Scaffold(
+          body: _buildUI(context, state),
+        );
+      },
+    );
+  }
+
+  Widget _buildUI(BuildContext context, CharacterState state) {
```
Comment (arrrtem22) [inline lib/feature/add_character/add_character_page.dart:59]: it's better to use Widgets, not helper methods
https://www.youtube.com/watch?v=IOyq-eTRhvo&ab_channel=Flutter

File: lib/feature/add_character/add_character_page.dart [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      body: BlocProvider(
+        create: (context) => CharacterCubit(),
+        child: AddCharacterForm(),
+      ),
+    );
+  }
+}
+
+class AddCharacterForm extends StatelessWidget {
+  final TextEditingController _nameController = TextEditingController();
```
Comment (arrrtem22) [inline lib/feature/add_character/add_character_page.dart]: when we work with controllers and stuff we should use stful widgets

Comment (arrrtem22) [reply lib/feature/add_character/add_character_page.dart]: it'll work in both cases but with stless we can have some memory leaks in the future

File: lib/feature/add_character/add_character_page.dart [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+          Text(
+            'Error: $error',
+            style: const TextStyle(fontSize: 18, color: Colors.red),
+          ),
+          const SizedBox(height: 10),
+          ElevatedButton(
+            onPressed: () {
+              context
+                  .read<CharacterCubit>()
+                  .addCharacter(_nameController.text.trim());
```
Comment (arrrtem22) [inline lib/feature/add_character/add_character_page.dart]: can we move trim to cubit?

File: lib/feature/add_character/cubit/add_character_state.dart [hunk 8]
```diff
@@ -0,0 +1,10 @@
+abstract class CharacterState {}
+
+class CharacterInitial extends CharacterState {}
+
+class CharacterAddedSuccess extends CharacterState {}
+
+class CharacterAddingError extends CharacterState {
```
Comment (arrrtem22) [inline lib/feature/add_character/cubit/add_character_state.dart:12]: why didn't you use state from mason?

File: lib/feature/home/home_page.dart [hunk 9]
```diff
@@ -69,6 +70,17 @@ class _LargeHomeView extends StatelessWidget {
                 child: state.selected != null
                     ? DetailView(character: state.selected!)
                     : const Center(child: Text('Select character'))),
+            FloatingActionButton(
+              onPressed: () {
+                Navigator.push(
```
Comment (arrrtem22) [inline lib/feature/home/home_page.dart:75]: use go_router for navigation

File: ios/Podfile.lock (hunk 1 above)
Comment (RomanDevelop) [reply ios/Podfile.lock:56]: Ok
I still struggle with understanding CocoaPods, and honestly, I haven't noticed how I changed the config files

File: lib/common/network/api/characters_api.dart (hunk 2 above)
Comment (RomanDevelop) [reply lib/common/network/api/characters_api.dart:24]: Снимок экрана 2023-12-23 в 18 05 54

File: lib/common/network/fake_api_manager.dart (hunk 3 above)
Comment (RomanDevelop) [reply lib/common/network/fake_api_manager.dart]: done)

File: ios/Podfile.lock (hunk 1 above)
Comment (arrrtem22) [reply ios/Podfile.lock:56]: if I have some problems with ios build, I usually run a few commands:
flutter clean
flutter pub get
cd ios && pod install --repo-update

File: lib/common/network/api/characters_api.dart (hunk 2 above)
Comment (arrrtem22) [reply lib/common/network/api/characters_api.dart:24]: yes, but I don't see changes, didn't you commit it?

Comment (arrrtem22) [reply lib/common/network/api/characters_api.dart:24]: you can copy your initial array from getCharacters method

Comment (arrrtem22) [reply lib/common/network/api/characters_api.dart:24]:
```suggestion
List<Character> fakeCharacters = [
      const Character(
        title: 'Fake Character 1',
        imageUrl: 'https://via.placeholder.com/600/92c952',
        description: 'This is a fake character description.',
      ),
      const Character(
        title: 'Fake Character 2',
        imageUrl: 'https://via.placeholder.com/600/7286a7',
        description: 'Another fake character description.',
      ),
    ]; 
```
